crypto/dh_extra/dh_test.cc

```diff
@@ -1051,3 +1051,110 @@ TEST(DHTest, DHCheckForStandardParams) {
   ASSERT_TRUE(DH_check(dh2.get(), &flags));
   EXPECT_EQ(flags, 0);
 }
+
+TEST(DHTest, DHMarshalPubKey) {
+  const char* dh512_pem =
+    "-----BEGIN DH PARAMETERS-----\n"
+    "MEYCQQDqvLe5oX3p+Dw8T7NWG7nlWVFK58Ev74xvxYH72DC4kqfPEFPvNnCpFoRB\n"
+    "RdxOz7DZ6JO/GxobSRyAAI766+GDAgEC\n"
+    "-----END DH PARAMETERS-----";
+  const uint64_t encoded_g = 2;
+  const char encoded_p_dec_str[] = "12294183602774786812319504504704470077603616440910559765086569005477513835495488680341310019770549315448633656928525381740662262980129138358936697450520963";
+
+  bssl::UniquePtr<EVP_PKEY> epkey_dh_params(nullptr);
+  {
+    const size_t pem_len = OPENSSL_strnlen(dh512_pem, 1024);
+    bssl::UniquePtr<BIO> in_bio(BIO_new_mem_buf(dh512_pem, pem_len));
+    ASSERT_TRUE(in_bio);
+
+    epkey_dh_params.reset(PEM_read_bio_Parameters(in_bio.get(), nullptr));
```

Comment:
Nit: We could just do
```suggestion
    bssl::UniquePtr<EVP_PKEY> epkey_dh_params(PEM_read_bio_Parameters(in_bio.get(), nullptr));
```

Another nit: Is there a particular reason for all these scope brackets in this test? They do seem a bit redundant.

Reply:
`epkey_dh_params` needs to be declared outside of the currently block/brackets so it's accessible by subsequent logic.

Reply:
> Another nit: Is there a particular reason for all these scope brackets in this test? They do seem a bit redundant.

I like to segment large test functions like this into smaller self-contained blocks. This reduces the "cognitive overhead" when reading the code, and allows for the use of simple variable names like `in_bio` and `out_bio`.